### PR TITLE
MM-52638- Last admin cannot be demoted

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Currently, the last admin can demote themselves resulting in no Admin users. This PR will check when an admin is demoted that other admin users exist.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52638

#### Release Note
```release-note
Last admin can no longer be demoted.
```

### CODE REVIEW

File: server/channels/app/user_test.go [hunk 1]
```diff
@@ -1687,6 +1687,45 @@ func TestUpdateUserRolesWithUser(t *testing.T) {
...
+	t.Run("Cannot remove if only admin", func(t *testing.T) {
+		// Attempt to downgrade sysadmin.
+		user, err := th.App.UpdateUserRolesWithUser(th.Context, th.SystemAdminUser, model.SystemUserRoleId, false)
+		require.NotNil(t, err)
+		require.Nil(t, user)
+	})
+
+	t.Run("Cannot remove if only non-Bot admin", func(t *testing.T) {
+		bot := th.CreateBot()
+		th.App.UpdateUserRoles(th.Context, bot.UserId, model.SystemUserRoleId+" "+model.SystemAdminRoleId, false)
```
Comment (isacikgoz) [inline server/channels/app/user_test.go:1712]: very nit but and I know this is a common pattern in our tests but if this fails, we shouldn't continue to the tests. Maybe worth checking the error.

**Review by isacikgoz — APPROVED** [review]
LGTM, just a small thing.

File: server/channels/app/user_test.go (hunk 1 above)
Comment (sbishel) [reply server/channels/app/user_test.go]: Done

**Comment by lindy65** [comment]
@sbishel, when I try to demote the last sys admin on the spinwick test server, I see a message "Unable to save roles." Confirming this is the correct message we want to show?

image

**Comment by sbishel** [comment]
Thanks @lindy65 - Yes, that is expected, as the modal displays a generic error on any error received.
https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/components/admin_console/manage_roles_modal/manage_roles_modal.tsx#L154
The error message `Cannot demote last System Admin.` does get logged.

**Review by lindy65 — APPROVED** [review]
Thanks for confirming @sbishel - looks good to me then 👍

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Pull request merged**